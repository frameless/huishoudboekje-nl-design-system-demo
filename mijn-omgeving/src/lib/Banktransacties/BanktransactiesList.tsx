import {Box, Stack, Text} from "@chakra-ui/react";
import "@utrecht/components/dist/heading-1/bem.css";
import "@utrecht/components/dist/table/bem.css";
import React from "react";
import {useTranslation} from "react-i18next";
import d from "../utils/dayjs";
import BanktransactieListItem from "./BanktransactieListItem";
import {Banktransactie} from "../../generated/graphql";
import Divider from "@gemeente-denhaag/divider";

const BanktransactiesList: React.FC<{ transacties: Banktransactie[] }> = ({transacties}) => {
	const {t} = useTranslation();

	const bt = transacties.reduce((result, t) => {
		const trDateAsString = d(t.transactiedatum).format("YYYY-MM-DD");
		return {
			...result,
			[trDateAsString]: [
				...(result[trDateAsString] || []),
				t,
			],
		};
	}, {});

	const dateString = (date: Date): string => {
		const _date = d(date).startOf("day");
		const today = d().startOf("day");

		if (_date.isSame(today)) {
			return t("date.today");
		}

		if (_date.isSame(today.subtract(1, "day"))) {
			return t("date.yesterday");
		}

		const format = _date.year() !== d().year() ? "dddd D MMMM YYYY" : "dddd D MMMM";
		return d(date).format(format);
	};

	return (<>
		{transacties.length > 0 ? (
			<Stack pt={4}>
				{Object.keys(bt).map((transactionDate, i) => {
					return (
						<Stack key={i}>
							<Box>
								<Text color={"gray"} fontSize={"sm"}>{dateString(d(transactionDate, "YYYY-MM-DD").toDate())}</Text>
							</Box>
							<Stack>
								{bt[transactionDate].map((transactie, i) => {
									return <BanktransactieListItem transactie={transactie} key={i} />;
								})}
							</Stack>
							<Divider orientation={"horizontal"} />
						</Stack>
					);
				})}
			</Stack>

		) : (
			<Text>{t("message.zeroTransactions")}</Text>
		)};
	</>);
};

export default BanktransactiesList;