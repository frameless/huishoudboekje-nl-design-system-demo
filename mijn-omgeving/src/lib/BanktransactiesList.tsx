import "@utrecht/components/dist/heading-1/bem.css";
import "@utrecht/components/dist/table/bem.css";
import React from "react";
import BanktransactieListItem from "./BanktransactieListItem";
import {Banktransactie} from "./generated/graphql";

const BanktransactiesList: React.FC<{transacties: Banktransactie[]}> = ({transacties}) => {
	return (<>
		<h1 className={"utrecht-heading-1"}>Uw Huishoudboekje</h1>

		{transacties.length > 0 ? (
			<table className={"utrecht-table"} style={{width: "100%"}}>
				<thead style={{
					textAlign: "left",
				}}>
					<tr>
						<th>Datum</th>
						<th>Tegenrekening</th>
						<th>Bedrag</th>
					</tr>
				</thead>
				<tbody>
					{transacties.map(transactie => {
						return <BanktransactieListItem transactie={transactie} />;
					})}
				</tbody>
			</table>
		) : (
			<span>Er zijn geen transacties gevonden.</span>
		)}
	</>);
};

export default BanktransactiesList;