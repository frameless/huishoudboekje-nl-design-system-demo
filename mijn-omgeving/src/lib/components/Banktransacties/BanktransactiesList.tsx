import {Box, Stack, Text} from "@chakra-ui/react";
import Divider from "@gemeente-denhaag/divider";
import "@utrecht/components/dist/heading-1/bem.css";
import "@utrecht/components/dist/table/bem.css";
import React from "react";
import {Banktransactie} from "../../../generated/graphql";
import {dateString} from "../../utils/dateFormat";
import d from "../../utils/dayjs";
import BanktransactieListItem from "./BanktransactieListItem";

const BanktransactiesList: React.FC<{ transacties: Banktransactie[] }> = ({transacties}) => {
	const bt = transacties.reduce((result, t) => {
		const trDateAsString = d(t.transactiedatum).format("YYYY-MM-DD");
		return {
			...result,
			[trDateAsString]: [
				...(result[trDateAsString] || []),
				t,
			],
		};
	}, {});

	return (
		<Stack>
			{Object.keys(bt).map((transactionDate, i) => {
				return (
					<Stack key={i}>
						<Box>
							<Text color={"gray"} fontSize={"sm"}>{dateString(d(transactionDate, "YYYY-MM-DD").toDate())}</Text>
						</Box>
						<Stack>
							{bt[transactionDate].map((transactie, i) => {
								return <BanktransactieListItem transactie={transactie} key={i} />;
							})}
						</Stack>
						<Divider orientation={"horizontal"} />
					</Stack>
				);
			})}
		</Stack>
	);
};

export default BanktransactiesList;