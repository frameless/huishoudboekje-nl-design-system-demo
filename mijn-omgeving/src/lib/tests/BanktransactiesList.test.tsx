import {act} from "@testing-library/react";
import {createMemoryHistory} from "history";
import React from "react";
import {render, unmountComponentAtNode} from "react-dom";
import {Router} from "react-router-dom";
import BanktransactiesList from "../components/Banktransacties/BanktransactiesList";
import banktransacties from "./data/banktransacties.json";

let container: HTMLDivElement | null = null;

beforeAll(() => {
	window.matchMedia = (query) => ({
		matches: false,
		media: query,
		onchange: null,
		addListener: jest.fn(),
		removeListener: jest.fn(),
		addEventListener: jest.fn(),
		removeEventListener: jest.fn(),
		dispatchEvent: jest.fn(),
	});
});

beforeEach(() => {
	container = document.createElement("div");
	document.body.appendChild(container);
});

afterEach(() => {
	unmountComponentAtNode(container!);
	container!.remove();
	container = null;
});

describe("Banktransacties List", () => {

	it("should show null transactions", function() {
		act(() => {
			render(<BanktransactiesList transacties={[]} />, container);
		});

		const element = container!.querySelector("div");
		expect(element).not.toBeNull();
		expect(element!.children.length).toEqual(0);
	});

	it("should render transactions", function() {
		const history = createMemoryHistory();
		act(() => {
			render((
				<Router location={history.location} navigator={history}>
					<BanktransactiesList transacties={banktransacties} />
				</Router>
			), container);
		});

		const element = container!.querySelector("div");
		expect(element).toMatchSnapshot();
	});
});
