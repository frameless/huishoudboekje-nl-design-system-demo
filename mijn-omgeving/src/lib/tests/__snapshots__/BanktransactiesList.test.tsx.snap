// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banktransacties List should render transactions 1`] = `
<div
  class="chakra-stack css-n21gh5"
>
  <div
    class="chakra-stack css-n21gh5"
  >
    <div
      class="css-0"
    >
      <p
        class="chakra-text css-dv4ab0"
      >
        zaterdag 25 december 2021
      </p>
    </div>
    <div
      class="chakra-stack css-n21gh5"
    >
      <div
        class="chakra-stack css-1oeb4ru"
      >
        <div
          class="css-0"
        >
          <p
            class="chakra-text css-0"
          >
            Albert Heijn B.V.
          </p>
        </div>
        <div
          class="css-0"
        >
          <div
            class="chakra-stack css-84zodg"
          >
            <p
              class="chakra-text css-1yniq95"
            >
              € 765,32
            </p>
            <a
              class=""
              href="/banktransacties/252"
            >
              <svg
                aria-hidden="true"
                class="denhaag-icon"
                fill="none"
                focusable="false"
                height="1em"
                shape-rendering="auto"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9.293 18.707a1 1 0 010-1.414L14.586 12 9.293 6.707a1 1 0 011.414-1.414l6 6a1 1 0 010 1.414l-6 6a1 1 0 01-1.414 0z"
                  fill="currentColor"
                />
              </svg>
            </a>
          </div>
        </div>
      </div>
    </div>
    <hr
      class="MuiDivider-root denhaag-divider"
      role="presentation"
    />
  </div>
  <div
    class="chakra-stack css-n21gh5"
  >
    <div
      class="css-0"
    >
      <p
        class="chakra-text css-dv4ab0"
      >
        maandag 20 december 2021
      </p>
    </div>
    <div
      class="chakra-stack css-n21gh5"
    >
      <div
        class="chakra-stack css-1oeb4ru"
      >
        <div
          class="css-0"
        >
          <p
            class="chakra-text css-0"
          >
            Belastingdienst
          </p>
        </div>
        <div
          class="css-0"
        >
          <div
            class="chakra-stack css-84zodg"
          >
            <p
              class="chakra-text css-1yniq95"
            >
              € 391,00
            </p>
            <a
              class=""
              href="/banktransacties/180"
            >
              <svg
                aria-hidden="true"
                class="denhaag-icon"
                fill="none"
                focusable="false"
                height="1em"
                shape-rendering="auto"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9.293 18.707a1 1 0 010-1.414L14.586 12 9.293 6.707a1 1 0 011.414-1.414l6 6a1 1 0 010 1.414l-6 6a1 1 0 01-1.414 0z"
                  fill="currentColor"
                />
              </svg>
            </a>
          </div>
        </div>
      </div>
      <div
        class="chakra-stack css-1oeb4ru"
      >
        <div
          class="css-0"
        >
          <p
            class="chakra-text css-0"
          >
            Belastingdienst
          </p>
        </div>
        <div
          class="css-0"
        >
          <div
            class="chakra-stack css-84zodg"
          >
            <p
              class="chakra-text css-1yniq95"
            >
              € 107,00
            </p>
            <a
              class=""
              href="/banktransacties/300"
            >
              <svg
                aria-hidden="true"
                class="denhaag-icon"
                fill="none"
                focusable="false"
                height="1em"
                shape-rendering="auto"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9.293 18.707a1 1 0 010-1.414L14.586 12 9.293 6.707a1 1 0 011.414-1.414l6 6a1 1 0 010 1.414l-6 6a1 1 0 01-1.414 0z"
                  fill="currentColor"
                />
              </svg>
            </a>
          </div>
        </div>
      </div>
      <div
        class="chakra-stack css-1oeb4ru"
      >
        <div
          class="css-0"
        >
          <p
            class="chakra-text css-0"
          >
            Belastingdienst
          </p>
        </div>
        <div
          class="css-0"
        >
          <div
            class="chakra-stack css-84zodg"
          >
            <p
              class="chakra-text css-1yniq95"
            >
              € 322,00
            </p>
            <a
              class=""
              href="/banktransacties/144"
            >
              <svg
                aria-hidden="true"
                class="denhaag-icon"
                fill="none"
                focusable="false"
                height="1em"
                shape-rendering="auto"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9.293 18.707a1 1 0 010-1.414L14.586 12 9.293 6.707a1 1 0 011.414-1.414l6 6a1 1 0 010 1.414l-6 6a1 1 0 01-1.414 0z"
                  fill="currentColor"
                />
              </svg>
            </a>
          </div>
        </div>
      </div>
    </div>
    <hr
      class="MuiDivider-root denhaag-divider"
      role="presentation"
    />
  </div>
  <div
    class="chakra-stack css-n21gh5"
  >
    <div
      class="css-0"
    >
      <p
        class="chakra-text css-dv4ab0"
      >
        dinsdag 7 december 2021
      </p>
    </div>
    <div
      class="chakra-stack css-n21gh5"
    >
      <div
        class="chakra-stack css-1oeb4ru"
      >
        <div
          class="css-0"
        >
          <p
            class="chakra-text css-0"
          >
            GEMEENTE UTRECHT
          </p>
        </div>
        <div
          class="css-0"
        >
          <div
            class="chakra-stack css-84zodg"
          >
            <p
              class="chakra-text css-1yniq95"
            >
              € 147,84
            </p>
            <a
              class=""
              href="/banktransacties/36"
            >
              <svg
                aria-hidden="true"
                class="denhaag-icon"
                fill="none"
                focusable="false"
                height="1em"
                shape-rendering="auto"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9.293 18.707a1 1 0 010-1.414L14.586 12 9.293 6.707a1 1 0 011.414-1.414l6 6a1 1 0 010 1.414l-6 6a1 1 0 01-1.414 0z"
                  fill="currentColor"
                />
              </svg>
            </a>
          </div>
        </div>
      </div>
    </div>
    <hr
      class="MuiDivider-root denhaag-divider"
      role="presentation"
    />
  </div>
  <div
    class="chakra-stack css-n21gh5"
  >
    <div
      class="css-0"
    >
      <p
        class="chakra-text css-dv4ab0"
      >
        zaterdag 4 december 2021
      </p>
    </div>
    <div
      class="chakra-stack css-n21gh5"
    >
      <div
        class="chakra-stack css-1oeb4ru"
      >
        <div
          class="css-0"
        >
          <p
            class="chakra-text css-0"
          >
            Portaal
          </p>
        </div>
        <div
          class="css-0"
        >
          <div
            class="chakra-stack css-84zodg"
          >
            <p
              class="chakra-text css-jo3ubs"
            >
              € -647,35
            </p>
            <a
              class=""
              href="/banktransacties/72"
            >
              <svg
                aria-hidden="true"
                class="denhaag-icon"
                fill="none"
                focusable="false"
                height="1em"
                shape-rendering="auto"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9.293 18.707a1 1 0 010-1.414L14.586 12 9.293 6.707a1 1 0 011.414-1.414l6 6a1 1 0 010 1.414l-6 6a1 1 0 01-1.414 0z"
                  fill="currentColor"
                />
              </svg>
            </a>
          </div>
        </div>
      </div>
    </div>
    <hr
      class="MuiDivider-root denhaag-divider"
      role="presentation"
    />
  </div>
  <div
    class="chakra-stack css-n21gh5"
  >
    <div
      class="css-0"
    >
      <p
        class="chakra-text css-dv4ab0"
      >
        woensdag 1 december 2021
      </p>
    </div>
    <div
      class="chakra-stack css-n21gh5"
    >
      <div
        class="chakra-stack css-1oeb4ru"
      >
        <div
          class="css-0"
        >
          <p
            class="chakra-text css-0"
          >
            ENGIE Nederland RETAIL B.V.
          </p>
        </div>
        <div
          class="css-0"
        >
          <div
            class="chakra-stack css-84zodg"
          >
            <p
              class="chakra-text css-jo3ubs"
            >
              € -132,00
            </p>
            <a
              class=""
              href="/banktransacties/108"
            >
              <svg
                aria-hidden="true"
                class="denhaag-icon"
                fill="none"
                focusable="false"
                height="1em"
                shape-rendering="auto"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9.293 18.707a1 1 0 010-1.414L14.586 12 9.293 6.707a1 1 0 011.414-1.414l6 6a1 1 0 010 1.414l-6 6a1 1 0 01-1.414 0z"
                  fill="currentColor"
                />
              </svg>
            </a>
          </div>
        </div>
      </div>
      <div
        class="chakra-stack css-1oeb4ru"
      >
        <div
          class="css-0"
        >
          <p
            class="chakra-text css-0"
          >
            F. de Jager
          </p>
        </div>
        <div
          class="css-0"
        >
          <div
            class="chakra-stack css-84zodg"
          >
            <p
              class="chakra-text css-jo3ubs"
            >
              € -939,94
            </p>
            <a
              class=""
              href="/banktransacties/216"
            >
              <svg
                aria-hidden="true"
                class="denhaag-icon"
                fill="none"
                focusable="false"
                height="1em"
                shape-rendering="auto"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9.293 18.707a1 1 0 010-1.414L14.586 12 9.293 6.707a1 1 0 011.414-1.414l6 6a1 1 0 010 1.414l-6 6a1 1 0 01-1.414 0z"
                  fill="currentColor"
                />
              </svg>
            </a>
          </div>
        </div>
      </div>
    </div>
    <hr
      class="MuiDivider-root denhaag-divider"
      role="presentation"
    />
  </div>
  <div
    class="chakra-stack css-n21gh5"
  >
    <div
      class="css-0"
    >
      <p
        class="chakra-text css-dv4ab0"
      >
        donderdag 25 november 2021
      </p>
    </div>
    <div
      class="chakra-stack css-n21gh5"
    >
      <div
        class="chakra-stack css-1oeb4ru"
      >
        <div
          class="css-0"
        >
          <p
            class="chakra-text css-0"
          >
            Albert Heijn B.V.
          </p>
        </div>
        <div
          class="css-0"
        >
          <div
            class="chakra-stack css-84zodg"
          >
            <p
              class="chakra-text css-1yniq95"
            >
              € 765,32
            </p>
            <a
              class=""
              href="/banktransacties/251"
            >
              <svg
                aria-hidden="true"
                class="denhaag-icon"
                fill="none"
                focusable="false"
                height="1em"
                shape-rendering="auto"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9.293 18.707a1 1 0 010-1.414L14.586 12 9.293 6.707a1 1 0 011.414-1.414l6 6a1 1 0 010 1.414l-6 6a1 1 0 01-1.414 0z"
                  fill="currentColor"
                />
              </svg>
            </a>
          </div>
        </div>
      </div>
    </div>
    <hr
      class="MuiDivider-root denhaag-divider"
      role="presentation"
    />
  </div>
  <div
    class="chakra-stack css-n21gh5"
  >
    <div
      class="css-0"
    >
      <p
        class="chakra-text css-dv4ab0"
      >
        zaterdag 20 november 2021
      </p>
    </div>
    <div
      class="chakra-stack css-n21gh5"
    >
      <div
        class="chakra-stack css-1oeb4ru"
      >
        <div
          class="css-0"
        >
          <p
            class="chakra-text css-0"
          >
            Belastingdienst
          </p>
        </div>
        <div
          class="css-0"
        >
          <div
            class="chakra-stack css-84zodg"
          >
            <p
              class="chakra-text css-1yniq95"
            >
              € 322,00
            </p>
            <a
              class=""
              href="/banktransacties/143"
            >
              <svg
                aria-hidden="true"
                class="denhaag-icon"
                fill="none"
                focusable="false"
                height="1em"
                shape-rendering="auto"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9.293 18.707a1 1 0 010-1.414L14.586 12 9.293 6.707a1 1 0 011.414-1.414l6 6a1 1 0 010 1.414l-6 6a1 1 0 01-1.414 0z"
                  fill="currentColor"
                />
              </svg>
            </a>
          </div>
        </div>
      </div>
      <div
        class="chakra-stack css-1oeb4ru"
      >
        <div
          class="css-0"
        >
          <p
            class="chakra-text css-0"
          >
            Belastingdienst
          </p>
        </div>
        <div
          class="css-0"
        >
          <div
            class="chakra-stack css-84zodg"
          >
            <p
              class="chakra-text css-1yniq95"
            >
              € 107,00
            </p>
            <a
              class=""
              href="/banktransacties/299"
            >
              <svg
                aria-hidden="true"
                class="denhaag-icon"
                fill="none"
                focusable="false"
                height="1em"
                shape-rendering="auto"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9.293 18.707a1 1 0 010-1.414L14.586 12 9.293 6.707a1 1 0 011.414-1.414l6 6a1 1 0 010 1.414l-6 6a1 1 0 01-1.414 0z"
                  fill="currentColor"
                />
              </svg>
            </a>
          </div>
        </div>
      </div>
      <div
        class="chakra-stack css-1oeb4ru"
      >
        <div
          class="css-0"
        >
          <p
            class="chakra-text css-0"
          >
            Belastingdienst
          </p>
        </div>
        <div
          class="css-0"
        >
          <div
            class="chakra-stack css-84zodg"
          >
            <p
              class="chakra-text css-1yniq95"
            >
              € 391,00
            </p>
            <a
              class=""
              href="/banktransacties/179"
            >
              <svg
                aria-hidden="true"
                class="denhaag-icon"
                fill="none"
                focusable="false"
                height="1em"
                shape-rendering="auto"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9.293 18.707a1 1 0 010-1.414L14.586 12 9.293 6.707a1 1 0 011.414-1.414l6 6a1 1 0 010 1.414l-6 6a1 1 0 01-1.414 0z"
                  fill="currentColor"
                />
              </svg>
            </a>
          </div>
        </div>
      </div>
    </div>
    <hr
      class="MuiDivider-root denhaag-divider"
      role="presentation"
    />
  </div>
  <div
    class="chakra-stack css-n21gh5"
  >
    <div
      class="css-0"
    >
      <p
        class="chakra-text css-dv4ab0"
      >
        zondag 7 november 2021
      </p>
    </div>
    <div
      class="chakra-stack css-n21gh5"
    >
      <div
        class="chakra-stack css-1oeb4ru"
      >
        <div
          class="css-0"
        >
          <p
            class="chakra-text css-0"
          >
            GEMEENTE UTRECHT
          </p>
        </div>
        <div
          class="css-0"
        >
          <div
            class="chakra-stack css-84zodg"
          >
            <p
              class="chakra-text css-1yniq95"
            >
              € 147,84
            </p>
            <a
              class=""
              href="/banktransacties/35"
            >
              <svg
                aria-hidden="true"
                class="denhaag-icon"
                fill="none"
                focusable="false"
                height="1em"
                shape-rendering="auto"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9.293 18.707a1 1 0 010-1.414L14.586 12 9.293 6.707a1 1 0 011.414-1.414l6 6a1 1 0 010 1.414l-6 6a1 1 0 01-1.414 0z"
                  fill="currentColor"
                />
              </svg>
            </a>
          </div>
        </div>
      </div>
    </div>
    <hr
      class="MuiDivider-root denhaag-divider"
      role="presentation"
    />
  </div>
  <div
    class="chakra-stack css-n21gh5"
  >
    <div
      class="css-0"
    >
      <p
        class="chakra-text css-dv4ab0"
      >
        donderdag 4 november 2021
      </p>
    </div>
    <div
      class="chakra-stack css-n21gh5"
    >
      <div
        class="chakra-stack css-1oeb4ru"
      >
        <div
          class="css-0"
        >
          <p
            class="chakra-text css-0"
          >
            Portaal
          </p>
        </div>
        <div
          class="css-0"
        >
          <div
            class="chakra-stack css-84zodg"
          >
            <p
              class="chakra-text css-jo3ubs"
            >
              € -647,35
            </p>
            <a
              class=""
              href="/banktransacties/71"
            >
              <svg
                aria-hidden="true"
                class="denhaag-icon"
                fill="none"
                focusable="false"
                height="1em"
                shape-rendering="auto"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9.293 18.707a1 1 0 010-1.414L14.586 12 9.293 6.707a1 1 0 011.414-1.414l6 6a1 1 0 010 1.414l-6 6a1 1 0 01-1.414 0z"
                  fill="currentColor"
                />
              </svg>
            </a>
          </div>
        </div>
      </div>
    </div>
    <hr
      class="MuiDivider-root denhaag-divider"
      role="presentation"
    />
  </div>
  <div
    class="chakra-stack css-n21gh5"
  >
    <div
      class="css-0"
    >
      <p
        class="chakra-text css-dv4ab0"
      >
        maandag 1 november 2021
      </p>
    </div>
    <div
      class="chakra-stack css-n21gh5"
    >
      <div
        class="chakra-stack css-1oeb4ru"
      >
        <div
          class="css-0"
        >
          <p
            class="chakra-text css-0"
          >
            ENGIE Nederland RETAIL B.V.
          </p>
        </div>
        <div
          class="css-0"
        >
          <div
            class="chakra-stack css-84zodg"
          >
            <p
              class="chakra-text css-jo3ubs"
            >
              € -132,00
            </p>
            <a
              class=""
              href="/banktransacties/107"
            >
              <svg
                aria-hidden="true"
                class="denhaag-icon"
                fill="none"
                focusable="false"
                height="1em"
                shape-rendering="auto"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9.293 18.707a1 1 0 010-1.414L14.586 12 9.293 6.707a1 1 0 011.414-1.414l6 6a1 1 0 010 1.414l-6 6a1 1 0 01-1.414 0z"
                  fill="currentColor"
                />
              </svg>
            </a>
          </div>
        </div>
      </div>
      <div
        class="chakra-stack css-1oeb4ru"
      >
        <div
          class="css-0"
        >
          <p
            class="chakra-text css-0"
          >
            F. de Jager
          </p>
        </div>
        <div
          class="css-0"
        >
          <div
            class="chakra-stack css-84zodg"
          >
            <p
              class="chakra-text css-jo3ubs"
            >
              € -939,94
            </p>
            <a
              class=""
              href="/banktransacties/215"
            >
              <svg
                aria-hidden="true"
                class="denhaag-icon"
                fill="none"
                focusable="false"
                height="1em"
                shape-rendering="auto"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9.293 18.707a1 1 0 010-1.414L14.586 12 9.293 6.707a1 1 0 011.414-1.414l6 6a1 1 0 010 1.414l-6 6a1 1 0 01-1.414 0z"
                  fill="currentColor"
                />
              </svg>
            </a>
          </div>
        </div>
      </div>
    </div>
    <hr
      class="MuiDivider-root denhaag-divider"
      role="presentation"
    />
  </div>
  <div
    class="chakra-stack css-n21gh5"
  >
    <div
      class="css-0"
    >
      <p
        class="chakra-text css-dv4ab0"
      >
        maandag 25 oktober 2021
      </p>
    </div>
    <div
      class="chakra-stack css-n21gh5"
    >
      <div
        class="chakra-stack css-1oeb4ru"
      >
        <div
          class="css-0"
        >
          <p
            class="chakra-text css-0"
          >
            Albert Heijn B.V.
          </p>
        </div>
        <div
          class="css-0"
        >
          <div
            class="chakra-stack css-84zodg"
          >
            <p
              class="chakra-text css-1yniq95"
            >
              € 765,32
            </p>
            <a
              class=""
              href="/banktransacties/250"
            >
              <svg
                aria-hidden="true"
                class="denhaag-icon"
                fill="none"
                focusable="false"
                height="1em"
                shape-rendering="auto"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9.293 18.707a1 1 0 010-1.414L14.586 12 9.293 6.707a1 1 0 011.414-1.414l6 6a1 1 0 010 1.414l-6 6a1 1 0 01-1.414 0z"
                  fill="currentColor"
                />
              </svg>
            </a>
          </div>
        </div>
      </div>
    </div>
    <hr
      class="MuiDivider-root denhaag-divider"
      role="presentation"
    />
  </div>
  <div
    class="chakra-stack css-n21gh5"
  >
    <div
      class="css-0"
    >
      <p
        class="chakra-text css-dv4ab0"
      >
        woensdag 20 oktober 2021
      </p>
    </div>
    <div
      class="chakra-stack css-n21gh5"
    >
      <div
        class="chakra-stack css-1oeb4ru"
      >
        <div
          class="css-0"
        >
          <p
            class="chakra-text css-0"
          >
            Belastingdienst
          </p>
        </div>
        <div
          class="css-0"
        >
          <div
            class="chakra-stack css-84zodg"
          >
            <p
              class="chakra-text css-1yniq95"
            >
              € 322,00
            </p>
            <a
              class=""
              href="/banktransacties/142"
            >
              <svg
                aria-hidden="true"
                class="denhaag-icon"
                fill="none"
                focusable="false"
                height="1em"
                shape-rendering="auto"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9.293 18.707a1 1 0 010-1.414L14.586 12 9.293 6.707a1 1 0 011.414-1.414l6 6a1 1 0 010 1.414l-6 6a1 1 0 01-1.414 0z"
                  fill="currentColor"
                />
              </svg>
            </a>
          </div>
        </div>
      </div>
      <div
        class="chakra-stack css-1oeb4ru"
      >
        <div
          class="css-0"
        >
          <p
            class="chakra-text css-0"
          >
            Belastingdienst
          </p>
        </div>
        <div
          class="css-0"
        >
          <div
            class="chakra-stack css-84zodg"
          >
            <p
              class="chakra-text css-1yniq95"
            >
              € 107,00
            </p>
            <a
              class=""
              href="/banktransacties/298"
            >
              <svg
                aria-hidden="true"
                class="denhaag-icon"
                fill="none"
                focusable="false"
                height="1em"
                shape-rendering="auto"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9.293 18.707a1 1 0 010-1.414L14.586 12 9.293 6.707a1 1 0 011.414-1.414l6 6a1 1 0 010 1.414l-6 6a1 1 0 01-1.414 0z"
                  fill="currentColor"
                />
              </svg>
            </a>
          </div>
        </div>
      </div>
      <div
        class="chakra-stack css-1oeb4ru"
      >
        <div
          class="css-0"
        >
          <p
            class="chakra-text css-0"
          >
            Belastingdienst
          </p>
        </div>
        <div
          class="css-0"
        >
          <div
            class="chakra-stack css-84zodg"
          >
            <p
              class="chakra-text css-1yniq95"
            >
              € 391,00
            </p>
            <a
              class=""
              href="/banktransacties/178"
            >
              <svg
                aria-hidden="true"
                class="denhaag-icon"
                fill="none"
                focusable="false"
                height="1em"
                shape-rendering="auto"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9.293 18.707a1 1 0 010-1.414L14.586 12 9.293 6.707a1 1 0 011.414-1.414l6 6a1 1 0 010 1.414l-6 6a1 1 0 01-1.414 0z"
                  fill="currentColor"
                />
              </svg>
            </a>
          </div>
        </div>
      </div>
    </div>
    <hr
      class="MuiDivider-root denhaag-divider"
      role="presentation"
    />
  </div>
</div>
`;
